test/controllers/records_controller_test.rb

```diff
@@ -121,6 +121,20 @@ class RecordsControllerTest < ActionController::TestCase
       assert_redirected_to collection_path(id: 1, record_id: record.id)
     end
 
+    it 'displays a failure message when granule association fails' do
```

Comment:
looks good, just rename the test slight, maybe "displays a failure message when attempting to assoc a granule to a collection record that has been closed."

Reply:
Done!